include/ddwaf.h

```diff
@@ -188,6 +188,18 @@ ddwaf_handle ddwaf_init(const ddwaf_object *rule, const ddwaf_config* config);
  */
 void ddwaf_destroy(ddwaf_handle handle);
 
+/**
+ * ddwaf_required_addresses
+ *
+ * Get a list of required (root) addresses. The memory is owned by the user, each
+ * element and the array should be freed with 'free'.
```

Comment:
Having the client of the library free memory allocated by the library is generally not a good idea, because there is no guarantee that the free inside the library was resolved to the same free in the client. In practice, this is mostly a problem on windows.

In this case, this problem can be easily side-stepped and a more friendly interface provided. Just make the memory owned be object pointed to by `ddwaf_handle`. This object can be a `std::vector<const char *>` whose elements are obtained by `std::string::c_str`, which is NUL-terminated. Then the C interface can just call `data()` on the vector. The result will be valid for the lifetime of the ddwaf_handle, since the manifest shouldn't change.